metrics/firebase/docs/01_firestore_builds_aggregation_design.md

```diff
@@ -0,0 +1,116 @@
+    "BuildStatus.success": number,
+    "BuildStatus.failed": number,
+    "BuildStatus.unknown": number,
+    "BuildStatus.in-progress": number,
+  },
+  "startedAt": timestamp,
+}
+```
+
+| Field | Description |
+| --- | --- |
+| `projectId`   | An id of the project, the build is related. |
+| `buildStatuses` | The map which contains BuildStatuses with the count of created builds. |
+| `startedAt`   | Determines the day build is started and contains a timestamp in the UTC. |
+
+There are situations, where creating or updating the `builds_per_day` collection fails. If it happens, we should store the build on which the counter was failed to the specific collection - `failed_builds_per_day`.
+
+The collection's document has the following structure:
+
+```json
+{
+  "projectId": String,
+  "buildNumber": number,
+  "action": String
+}
+```
+
+| Field | Description |
+| --- | --- |
+| `projectId`   | An id of the project, the build is related. |
+| `buildNumber` | A number of the build. |
+| `action`   | The process, that is failed (e.g.g. onCreate trigger for the builds collection). |
+
+_**Note:** The collection contains the `projectId` and `buildNumber` to uniquely identify a single build in the `build` collection._
+
+Later, using the `projectId` and `buildNumber` we can fetch a build from the `build` collection with the latest status and update the `builds_per_day` collection.
+
+## Firestore security rules
+
+Once we have new collections, we have to add security rules for them.
+
+First, the `builds_per_day` collection, everyone can read the content but no one has access to create, update or delete collection documents.
+
+Thus, the following rules keep:
+
+- Access-related rules:
+  - not authenticated users **cannot** read, create, update, delete this document;
+  - authenticated users **can** read and **cannot** create, update, delete this document.
+
+Second, for the `failed_builds_per_day` collection - no one has access to read, create, update or delete collection documents.
+
+- Access-related rules:
+  - not authenticated users **cannot** read, create, update, delete this document;
+  - authenticated users **cannot** read, create, update, delete this document.
+
+## Firestore Cloud Functions
+
+When we've created collections and applied rules for them, we should create the Firestore Cloud Functions. In addition, we can [write these functions using the Dart programming language](https://github.com/platform-platform/monorepo/blob/master/metrics/firebase/docs/analysis/01_using_dart_in_the_firebase_cloud_functions.md).
+
+If we want to provide builds aggregations, we need to process calculations in reaction to added build. For this purpose, we should create the `onCreate` function trigger on the `build` collection.
+
+```dart
+functions['onAddedBuild'] = functions.firestore
+    .document('build/{buildId}')
+    .onCreate(onAddedBuildHandler)
+
+
+FutureOr<void> onAddedBuildHandler(DocumentSnapshot snapshot, EventContext context) {...}
```

Comment:
Please, avoid using the `FutureOr` as a return type. See [Effective Dart](https://dart.dev/guides/language/effective-dart/design#avoid-using-futureort-as-a-return-type) for more details. 
```suggestion
Future<void> onAddedBuildHandler(DocumentSnapshot snapshot, EventContext context) {...}
```